### PR TITLE
Don't run tests on deploy branch

### PR DESCRIPTION
**Pull request by snario**
Not sure if the other jobs which require it will automatically inherit this

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -352,7 +352,10 @@ jobs:
 workflows:
   statechannels:
     jobs:
-      - prepare
+      - prepare:
+          filters:
+            branches:
+              ignore: liam/noprepare
```
Comment (snario) [inline .circleci/config.yml:358]: Will change to `deploy` if tests pass

**Comment by kerzhner** [comment]
@snario I might be missing something, but don't you need to build monorepo packages that the xstate-wallet and w3t depend on (like nitro-protocol and wire-format) before building those packages? That is basically what `prepare` job does.

**Comment by snario** [comment]
I think that is happening here right now

https://github.com/statechannels/monorepo/blob/fb276ae49838a773d9cf3ccc2129f04f274c36da/.circleci/config.yml#L328-L328

Also, `deploy` runs in parallel to `prepare`

**Comment by snario** [comment]
`release-netlify-production` definitely need `prepare` actually, but not `release-hub-production`

**Comment by kerzhner** [comment]
Check out https://app.circleci.com/pipelines/github/statechannels/monorepo/5865/workflows/075b8ee1-459d-42ce-ab78-429ab198da8f to see the job depencdy graph of a `deploy` workflow. `release-hub-production` does not depend on prepare for the reason you pointed out. release-`netlify-production` depends on `prepare` for the reason I pointed out.

**Comment by snario** [comment]
You're right  -- the spirit of this PR was to not re-run the tests on `deploy`, let me change it to do that correctly

**Comment by kerzhner** [comment]
> You're right -- the spirit of this PR was to not re-run the tests on `deploy`, let me change it to do that correctly

Gotcha. I am of the opinion that we actually want to run test on the deploy workflow. What if the deploy branch contains a hotfix that was never included in master? In that case, the tests would never get run for that code change.

Also, the test are non-blocking for deploys, so the only downside is that we are burning circlci credits. But how often do we expect to deploy?

**Comment by snario** [comment]
> What if the deploy branch contains a hotfix that was never included in master?
I think we should open PRs against master in addition to committing hotfixes

**Comment by kerzhner** [comment]
So then master might contain commits A->B-H where H is a hotfix. `deploy` might contain commits A->H because B is not ready for deployment. What if H doesn't work without B? Tests might help us catch that.

**Comment by kerzhner** [comment]
My overall points are:
- It is not a given that tests have been run on the code being deployed.
- Tests are not blocking, so why not run them?

**Pull request closed without merging**